### ISSUE TITLE
LTE card issues when running tch-gui-unhide

### ISSUE THREAD

**Issue by willat8** [seud0nym/tch-gui-unhide#119]
On 18.1.c (DJA0230), and release 2022.11.30, I'm seeing these errors when running tch-gui-unhide (minifcation enabled). My LTE is disabled and the card is hidden.

```
190@21:44:48: Checking card visibility
190@21:44:57: Applying configured card sequence
mv: can't stat '/www/cards/010_lte.lp/003_lte.lp': Not a directory
mv: can't stat '/www/cards/010_lte.lp/010_lte.lp': Not a directory
190@21:44:59: Checking all configured cards exist
200@21:45:00: Add missing validateLXC post_helper function
200@21:45:00: Add specific error messages when table edits fail
300@21:45:00: Fix Basic view header
500@21:45:00: Checking power saving settings...
600@21:45:01: Adding or updating card background icons
600@21:45:01: Fix some responsive span widths
600@21:45:01: Add theme processing
610@21:45:04: Deploy theme files
610@21:45:05: Persisting theme settings
BLD@21:45:05: Adding version details to copyright
BLD@21:45:06: Auto-refreshing browser cache
800@21:45:07: Pre-minify fixes
800@21:45:07: Minifying GUI Lua code
lua: /usr/share/lua/llex.lua:129: [string]:2: malformed number
stack traceback:
	[C]: in function 'e'
	/usr/share/lua/llex.lua:129: in function 'errorline'
	/usr/share/lua/llex.lua:270: in function 'llex'
	/usr/share/lua/LuaSrcDiet.lua:427: in function 'process_file'
	/usr/share/lua/LuaSrcDiet.lua:571: in function 'do_files'
	/usr/share/lua/LuaSrcDiet.lua:659: in function 'main'
	/usr/share/lua/LuaSrcDiet.lua:667: in main chunk
	[C]: ?
800@21:45:07: - LuaSrcDiet failed for /www/cards/010_lte.lp
800@21:46:07: - Lua GUI code size reduced by 38% in 60 seconds
```

**Comment by seud0nym** [comment]
That is disappointing. I didn't think I had changed anything anywhere near that code. And those filenames in the "Not a directory" error are really weird.

A couple of questions:

- Which particular 18.1.c release (e.g. 18.1.c.0549-MR17-RB)? 
- Was this an upgrade to an existing installation, or a fresh install after RTFD?
  - If a clean install, what were the parameters?
  - If not, do you know what parameters were initially applied?

Thanks.

**Comment by willat8** [comment]
Yep, 18.1.c.0549-MR17-RB.

Fresh install after double reset button + re-root.

`-by -dy -fn -py -vy -my -a4 -cm -hn -in -ln -tc -Cs`

Then disabled mobile in LTE card.

Then ran tch-gui-unhide-cards and hid LTE card and several other cards.

Then swapped to bridged mode.

Then ran

```sh
opkg --force-overwrite install samba36-server
./tch-gui-unhide -x samba36-server
```

Which is when I noticed this error for the first time.

**Comment by seud0nym** [comment]
First try I couldn't reproduce.

What options did you use for de-telstra?

**Comment by willat8** [comment]
```
Keep Telstra AIR enabled                 -ka              No
  Keep CWMP installed                      -kc              No
  Keep default public authorized keys      -kk              No
  Keep Telstra APN check                   -kl              No
  Keep Telstra monitoring/data collection  -km              No
  Keep Telstra NTP servers                 -kn              No
  Keep Telstra QoS VoWiFi reclassify rules -kq              No
  Keep noexec on filesystems               -kx              No
  Host name                                -hsmbhost        smbhost
  Domain name                              -dlan            lan
  DNS servers                              -nc              Cloudflare
  NAT ALG Helpers                          -an              Disabled
  Content Sharing                          -cy              Enabled
  DECT Emission Mode                       -en              Disabled
  File Sharing                             -fy              Enabled
  DumaOS (Game Optimiser)                  -gn              Disabled
  Intercept Daemon                         -in              Disabled
  LED Logging                              -ln              Disabled
  EasyMesh Agent                           -mn              Disabled
  EasyMesh Controller                      -mn              Disabled
  EasyMesh BackHaul SSID                   -mn              Disabled
  MultiAP (EasyMesh) Vendor Extensions     -mn              Disabled
  Power Saving                             -pn              No
  Printer Sharing                          -rn              Disabled
  WAN Supervision                          -sn              Disabled
  Telephony                                -tn              Disabled
  UPnP                                     -un              Disabled
  WPS                                      -wn              Disabled
  RTFD root protection                     -Fy              Enabled
  Minimum memory mode                      -M               Yes
  Remove Guest Wi-Fi                       -G               Yes
  LAN IPv4 Address                         -I               Unchanged
  Configure opkg                           -o               Yes
   - opkg repository                       -O18             18
```

**Comment by seud0nym** [comment]
Thanks. Can you try running this:
```
./tch-gui-unhide --verbose
```

There will be a bit more logging than usual. I am just interested in the stuff where it fails.

**Comment by willat8** [comment]
Sure:

```
190@12:09:26: - Card Rule web.card_lte: card=010_lte.lp search=/www/cards/???_lte.lp found=/www/cards/003_lte.lp
/www/cards/010_lte.lp
190@12:09:26: - Renaming /www/cards/003_lte.lp
/www/cards/010_lte.lp to /www/cards/010_lte.lp
mv: can't stat '/www/cards/010_lte.lp/003_lte.lp': Not a directory
mv: can't stat '/www/cards/010_lte.lp/010_lte.lp': Not a directory
```

and

```
lua: /usr/share/lua/llex.lua:129: [string]:2: malformed number
stack traceback:
        [C]: in function 'e'
        /usr/share/lua/llex.lua:129: in function 'errorline'
        /usr/share/lua/llex.lua:270: in function 'llex'
        /usr/share/lua/LuaSrcDiet.lua:427: in function 'process_file'
        /usr/share/lua/LuaSrcDiet.lua:571: in function 'do_files'
        /usr/share/lua/LuaSrcDiet.lua:659: in function 'main'
        /usr/share/lua/LuaSrcDiet.lua:667: in main chunk
        [C]: ?
800@12:09:32: - LuaSrcDiet failed for /www/cards/010_lte.lp
```

`/www/cards/010_lte.lp` looks like

```lua
--pretranslated: do not change this file
ngx.print('<style>.card-label {height: 17px;margin-bottom:0px;float: left;padding-right: 5px;font-weight: 500;font-size: 16px;cursor: default;}.card-control {cursor: default;}</style>');local e=ngx
 height: 17px;margin-bottom:0px;height: 17px;margin-bottom:0px;height: 17px;margin-bottom:0px;gettext.textdomain('webui-mobiled')local l=require("web.lte-utils")local a=l.getContent("rpc.mobiled.DeviceNumberOfEntries")local a=tonumber(a.DeviceNumberOfEntries)if a and a>0 then
local result = utils.getContent("rpc.mobiled.DeviceNumberOfEntries")
local devices = tonumber(result.DeviceNumberOfEntries)
if devices and devices > 0 then
local result = utils.getContent("rpc.mobiled.DeviceNumberOfEntries")
local devices = tonumber(result.DeviceNumberOfEntries)
if devices and devices > 0 then
local result = utils.getContent("rpc.mobiled.DeviceNumberOfEntries")
local devices = tonumber(result.DeviceNumberOfEntries)
if devices and devices > 0 then
local a=require("web.ui_helper")local n=l.get_params()local d=e.ctx.session
local s=false
if d:hasAccess("/modals/lte-sim.lp")then
s=true
end
local t=require("datamodel")local i=t.get("uci.wansensing.global.enable","uci.wansensing.global.primarywanmode")local t={status="0",desc=T"Disabled"}local r=l.get_uci_device_path()local l=l.getContent(r.."enabled")if l['enabled']=="1"then
t.status="1"t.desc=T"Enabled"end
local l=nil
if d:hasAccess("/modals/lte-modal.lp")then
l="/modals/lte-modal.lp"end
e.print('<link href="/css/mobiled.css?18.1.c_2022.11.30@14:10" rel="stylesheet"><div class="span3" id="mobiletab"><div class="smallcard">',a.createCardHeader(n.card_title,l));e.print('<div id="signal-strength-indicator-small"><div class="absolute"><div class="bar-small bar-small1"></div><div class="bar-small bar-small2"></div><div class="bar-small bar-small3"></div><div class="bar-small bar-small4"></div><div class="bar-small bar-small5"></div></div></div><div class="content card_bg" data-bg-text="&#xf012;">');if i and i[1].value=="1"then
if i[2].value=="MOBILE"then
e.print(a.createSimpleLight("1",T("Mobile Only Mode enabled")));else
e.print(a.createSimpleLight("0",T("Mobile Only Mode disabled")));end
end
 $("#mobiletab p.subinfos").show();<div style="height: 17px;" data-bind="visible: unreadSMS().length > 0"><label class="card-label modal-link" data-toggle="modal" data-remote="modals/lte-sms.lp" data-id="lte-sms">'); ngx.print(T"Unread SMS"..":" ,'</label><div class="controls"><strong data-bind="text: unreadSMS"></strong></div></div>$("#mobiletab p.subinfos").show();<div style="height: 17px;" data-bind="visible: unreadSMS().length > 0"><label class="card-label modal-link" data-toggle="modal" data-remote="modals/lte-sms.lp" data-id="lte-sms">'); ngx.print(T"Unread SMS"..":" ,'</label><div class="controls"><strong data-bind="text: unreadSMS"></strong></div></div>$("#mobiletab p.subinfos").show();<div style="height: 17px;" data-bind="visible: unreadSMS().length > 0"><label class="card-label modal-link" data-toggle="modal" data-remote="modals/lte-sms.lp" data-id="lte-sms">'); ngx.print(T"Unread SMS"..":" ,'</label><div class="controls"><strong data-bind="text: unreadSMS"></strong></div></div>e.print(a.createSimpleLight(t.status,t.desc));e.print('<p class="subinfos" style="display:none;"><div style="height: 17px;" data-bind="visible: registrationStatus().length > 0"><label class="card-label">');e.print(T"Network"..":",'</label><div class="controls"><strong data-bind="text: registrationStatus"></strong></div></div><div style="height:17px;" data-bind="visible: mobileStatus().length > 0"><label class="card-label">');e.print(T"Status"..":",'</label><div class="controls"><strong data-bind="text: mobileStatus"></strong></div></div><div style="height:17px;" data-bind="visible: radioType().length > 0"><label class="card-label">');e.print(T"Radio Type"..":",'</label><div class="card-control"><strong data-bind="text: radioType"></strong></div></div><div style="height:17px;" data-bind="visible: signalQuality().length > 0"><label class="card-label">');e.print(T"Quality"..":",'</label><div class="controls"><strong data-bind="text: signalQuality"></strong></div></div><div style="height:17px;" data-bind="visible: deviceStatus().length > 0"><div class="controls"><strong data-bind="text: deviceStatus"></strong></div></div><div style="height: 17px;" data-bind="visible: unreadSMS().length > 0"><label class="card-label modal-link" data-toggle="modal" data-remote="modals/lte-sms.lp" data-id="lte-sms">');e.print(T"Unread SMS"..":",'</label><div class="controls"><strong data-bind="text: unreadSMS"></strong></div></div></p></div></div></div><script>var divs = $("#mobiletab .content").children("div");if(divs.length>0){var p=$("#mobiletab .content .subinfos");divs.appendTo(p);}$(document).ready(function () {var allowSimRedirect = ');e.print(s,';var LteCard = function () {var clearRefreshInterval = function () {clearInterval(refreshInterval);};var self = this;this.signalQuality = ko.observable("");this.radioType = ko.observable("");this.mobileStatus = ko.observable("");this.deviceStatus = ko.observable("");this.registrationStatus = ko.observable("");this.unreadSMS = ko.observable("");this.service_provider = ko.observable("");this.updateBars = function(bars) {$(\'#signal-strength-indicator-small .absolute\').children(\'.bar-small\').each(function (index) {if(index < bars) {$(this).addClass(\'bar-small-active\');} else {$(this).removeClass(\'bar-small-active\');}});};this.refresh = function () {$("#mobiletab p.subinfos").show();$.ajax({ type: "POST", url: "/ajax/mobiletab.lua?auto_update=true",dataType: "json", data: [tch.elementCSRFtoken()], success: function(data) {if(data.registration_status != undefined) {self.registrationStatus(data.registration_status);}if(data.unread_sms != undefined) {self.unreadSMS(data.unread_sms);}if(data.service_provider != undefined) {$("#Mobile_tab").text("Mobile ").append("<span style=\\"font-size:x-small\\">"+data.service_provider+"</span>");}if(data.bars != undefined) {self.updateBars(data.bars);}if(data.radio_interface != undefined) {self.radioType(data.radio_interface);}if(data.status != undefined) {self.mobileStatus(data.status);}if(data.signal_quality != undefined) {self.signalQuality(data.signal_quality);}if(data.no_device != undefined) {self.deviceStatus(data.no_device);}if(data.redirect_sim != undefined) {if(data.redirect_sim === true && allowSimRedirect) {$.each($(\'.settings[data-id="lte-modal"]\'), function () {$(this).attr(\'data-id\', \'lte-sim\');$(this).attr(\'data-remote\', \'/modals/lte-sim.lp\');});} else {$.each($(\'.settings[data-id="lte-sim"]\'), function () {$(this).attr(\'data-id\', \'lte-modal\');$(this).attr(\'data-remote\', \'/modals/lte-modal.lp\');});}}}, error: clearRefreshInterval });};var refreshInterval = setInterval(self.refresh, 20000);addRegisteredInterval(refreshInterval);self.refresh();};var lteCard = new LteCard();ko.applyBindings(lteCard, document.getElementById(\'mobiletab\'));});var div = document.querySelector("#mobiletab").querySelector(".header-title");var signal = document.querySelector("#signal-strength-indicator-small");div.parentNode.insertBefore(signal, div.nextSibling);</script>');end
<div style="height: 17px;" data-bind="visible: registrationStatus().length > 0"><label class="card-label">'); ngx.print(T"Network"..":",'</label><div class="controls"><strong data-bind="text: registrationStatus"></strong></div></div>this.registrationStatus = ko.observable("");this.unreadSMS = ko.observable("");this.service_provider = ko.observable("");if(data.registration_status != undefined) {self.registrationStatus(data.registration_status);}if(data.unread_sms != undefined) {self.unreadSMS(data.unread_sms);}if(data.service_provider != undefined) {$("#Mobile_tab").text("Mobile ").append("<span style=\\"font-size:x-small\\">"+data.service_provider+"</span>");}var divs = $("#mobiletab .content").children("div");if(divs.length>0){var p=$("#mobiletab .content .subinfos");divs.appendTo(p);}addRegisteredInterval(refreshInterval);<div style="height: 17px;" data-bind="visible: registrationStatus().length > 0"><label class="card-label">'); ngx.print(T"Network"..":",'</label><div class="controls"><strong data-bind="text: registrationStatus"></strong></div></div>this.registrationStatus = ko.observable("");this.unreadSMS = ko.observable("");this.service_provider = ko.observable("");if(data.registration_status != undefined) {self.registrationStatus(data.registration_status);}if(data.unread_sms != undefined) {self.unreadSMS(data.unread_sms);}if(data.service_provider != undefined) {$("#Mobile_tab").text("Mobile ").append("<span style=\\"font-size:x-small\\">"+data.service_provider+"</span>");}var divs = $("#mobiletab .content").children("div");if(divs.length>0){var p=$("#mobiletab .content .subinfos");divs.appendTo(p);}addRegisteredInterval(refreshInterval);<div style="height: 17px;" data-bind="visible: registrationStatus().length > 0"><label class="card-label">'); ngx.print(T"Network"..":",'</label><div class="controls"><strong data-bind="text: registrationStatus"></strong></div></div>this.registrationStatus = ko.observable("");this.unreadSMS = ko.observable("");this.service_provider = ko.observable("");if(data.registration_status != undefined) {self.registrationStatus(data.registration_status);}if(data.unread_sms != undefined) {self.unreadSMS(data.unread_sms);}if(data.service_provider != undefined) {$("#Mobile_tab").text("Mobile ").append("<span style=\\"font-size:x-small\\">"+data.service_provider+"</span>");}var divs = $("#mobiletab .content").children("div");if(divs.length>0){var p=$("#mobiletab .content .subinfos");divs.appendTo(p);}addRegisteredInterval(refreshInterval);end
end
end
```

**Comment by seud0nym** [comment]
Okay so the issue is that you have 2 lte cards. The question is how did that happen?

What does this return?
```
ls -l /rom/www/cards/*_lte.lp
```

**Comment by willat8** [comment]
`-rw-r--r--    1 root     root          4988 Sep  1  2020 /rom/www/cards/010_lte.lp`

**Comment by willat8** [comment]
I see this in `tch-gui-unhide-cards`, I wonder if it got confused when I disabled LTE and then hid the card with `tch-gui-unhide-cards`.

`NOTE: 'Mobile' card is only visible if supported by device and has not been disabled`

**Comment by seud0nym** [comment]
No, I don't think that tch-gui-unhide-cards is the problem. I could be wrong, but I just can't see it in the code.

Sorry to keep asking you for output. What does this return:
```
uci show web | grep lte.lp
```

**Comment by seud0nym** [comment]
Don't worry about that last output. There is some very early code (over 2 years old) around handling the change of numbering for the Mobile card. It would have been related to the way that tch-gui-unhide worked 2 years ago, which is slightly different to now. I have pushed up a test for you try:
```
./tch-gui-unhide -u && /tch-gui-unhide -y
```

**Comment by willat8** [comment]
That's working! No more errors :). Thanks.

**Comment by seud0nym** [comment]
Fixed in release 2022.12.21
